components/brave_vpn/brave_vpn_service.cc

```diff
@@ -1029,6 +1041,63 @@ void BraveVpnService::OnPrepareCredentialsPresentation(
 #endif
 }
 
+void BraveVpnService::InitP3A() {
+  p3a_timer_.Start(FROM_HERE, base::Hours(kP3AIntervalHours), this,
+                   &BraveVpnService::OnP3AInterval);
+  RecordP3A(false);
+}
+
+void BraveVpnService::RecordP3A(bool new_usage) {
+  if (new_usage) {
+    p3a_utils::RecordFeatureUsage(local_prefs_, prefs::kBraveVPNFirstUseTime,
+                                  prefs::kBraveVPNLastUseTime);
+  }
+  p3a_utils::RecordFeatureNewUserReturning(
+      local_prefs_, prefs::kBraveVPNFirstUseTime, prefs::kBraveVPNLastUseTime,
+      prefs::kBraveVPNUsedSecondDay, kNewUserReturningHistogramName);
+  p3a_utils::RecordFeatureDaysInMonthUsed(
+      local_prefs_, new_usage, prefs::kBraveVPNLastUseTime,
+      prefs::kBraveVPNDaysInMonthUsed, kDaysInMonthUsedHistogramName);
+  p3a_utils::RecordFeatureLastUsageTimeMetric(
+      local_prefs_, prefs::kBraveVPNLastUseTime, kLastUsageTimeHistogramName);
+}
+
+void BraveVpnService::RecordAndroidBackgroundP3A(int64_t session_start_time_ms,
```

Comment:
How about guarding this method with `BUILDFLAG(IS_ANDROID)` if it's only used on Android?